tests/integration/targets/ndo_synce_interface_policy/tasks/main.yml

```diff
@@ -0,0 +1,266 @@
+          - nm_update_synce_interface_policy.current.adminState == "enabled"
+          - nm_update_synce_interface_policy.current.syncStateMsgEnabled == true
+          - nm_update_synce_interface_policy.current.selectionInputEnabled == true
+          - nm_update_synce_interface_policy.current.srcPriority == 110
+          - nm_update_synce_interface_policy.current.waitToRestore == 12
+          - nm_update_synce_interface_policy_again is not changed
+          - nm_update_synce_interface_policy_again.previous.description == nm_update_synce_interface_policy_again.current.description == "changed_description"
+          - nm_update_synce_interface_policy_again.previous.adminState == nm_update_synce_interface_policy_again.current.adminState == "enabled"
+          - nm_update_synce_interface_policy_again.previous.syncStateMsgEnabled == nm_update_synce_interface_policy_again.current.syncStateMsgEnabled == true
+          - nm_update_synce_interface_policy_again.previous.selectionInputEnabled == nm_update_synce_interface_policy_again.current.selectionInputEnabled == true
+          - nm_update_synce_interface_policy_again.previous.srcPriority == nm_update_synce_interface_policy_again.current.srcPriority == 110
+          - nm_update_synce_interface_policy_again.previous.waitToRestore == nm_update_synce_interface_policy_again.current.waitToRestore == 12
+
+    # QUERY
+
+    - name: Create another syncE interface policy
+      cisco.mso.ndo_synce_interface_policy: &add_synce_interface_policy_2
+        <<: *add_synce_interface_policy
+        interface_policy: ansible_test_synce_interface_policy_2
+        description: "This is a test syncE interface policy"
+      register: nm_add_synce_interface_policy_2
+
+    - name: Query a syncE interface policy with template_name
+      cisco.mso.ndo_synce_interface_policy:
+        <<: *mso_info
+        template: ansible_fabric_policy_template
+        interface_policy: ansible_test_synce_interface_policy
+        state: query
+      register: query_one
+
+    - name: Query all syncE interface policy in the template
+      cisco.mso.ndo_synce_interface_policy:
+        <<: *mso_info
+        template: ansible_fabric_policy_template
+        state: query
+      register: query_all
+
+    - name: Verify query_one and query_all
+      assert:
+        that:
+          - query_all is not changed
+          - query_one is not changed
+          - query_all.current | length >= 2
+          - query_one.current.name == "ansible_test_synce_interface_policy"
+          - query_one.current.description == "changed_description"
+
+    # DELETE
+
+    - name: Delete a syncE interface policy (check mode)
+      cisco.mso.ndo_synce_interface_policy: &rm_synce_interface_policy
+        <<: *mso_info
+        template: ansible_fabric_policy_template
+        interface_policy: ansible_test_synce_interface_policy
+        state: absent
+      check_mode: true
+      register: cm_rm_synce_interface_policy
```

Comment:
Can you add the proposed value check for the check mode tests?

Reply:
Proposed value check will not be added now, since the existing modules with proposed gives the below error 
----> 'dict object' has no attribute 'proposed'

This will be fixed later.